app/models/invoice.rb

```diff
@@ -612,4 +615,15 @@ def notify_buyer_about_payment
       InvoiceMessenger.successfully_charged(self).deliver
     end
   end
+
+  private
+
+  # Allowed range for the invoice period is:
+  # - from: creation date of the provider
+  # - to: 12 months from now
+  def period_range_valid?
+    return if self[:period]&.between?(provider_account.created_at.to_date.beginning_of_month, Time.zone.now + 12.months)
```

Comment:
Probably it doesn't matter here but letting timezones not specified is a red flag for me. Given the time period is generous here though, I guess a time zone will be insignificant.